pebbles/views/namespaced_keyvalues.py

```diff
@@ -0,0 +1,123 @@
+from flask.ext.restful import fields, marshal_with, reqparse
+from flask import abort, Blueprint
+
+import logging
+import time
+
+from pebbles.models import db, NamespacedKeyValue
+from pebbles.forms import NamespacedKeyValueForm
+from pebbles.server import restful
+from pebbles.views.commons import auth
+from pebbles.utils import requires_admin
+
+namespaced_keyvalues = Blueprint('namespaced_keyvalues', __name__)
+
+
+namespace_fields = {
+    'id': fields.String(attribute='id'),
+    'namespace': fields.String,
+    'key': fields.String,
+    'value': fields.Raw,
+    'created': fields.Float,
+    'updated': fields.Float
+}
+
+
+class NamespacedKeyValueList(restful.Resource):
+    parser = reqparse.RequestParser()
+    parser.add_argument('namespace', type=str)
+    parser.add_argument('key', type=str)
+
+    @auth.login_required
+    @requires_admin
+    @marshal_with(namespace_fields)
+    def get(self):
+        args = self.parser.parse_args()
+        namespaced_query = NamespacedKeyValue.query
+        if args.get('namespace'):
+            namespaced_query = namespaced_query.filter_by(namespace=args.namespace)
+        if args.get('key'):
+            namespaced_query = namespaced_query.filter_by(key=args.key)
+
+        return namespaced_query.all()
+
+    @auth.login_required
+    @requires_admin
+    def post(self):
+        form = NamespacedKeyValueForm()
+
+        if not form.validate_on_submit():
+            logging.warn(form.errors)
+            logging.warn("validation error on creating namespaced key value data")
+            return form.errors, 422
+
+        namespace = form.namespace.data
+        key = form.key.data
+        value = form.value.data
+
+        ns_check = NamespacedKeyValue.query.filter_by(namespace=namespace, key=key).first()
+        if ns_check:
+            logging.warn("a combination of namespace %s with key %s already exists" % (namespace, key))
+            abort(422)
+        namespaced_keyvalue = NamespacedKeyValue(namespace, key)
+        namespaced_keyvalue.value = value
+
+        curr_ts = time.time()
+        namespaced_keyvalue.created_ts = curr_ts
+        namespaced_keyvalue.updated_ts = curr_ts
+
+        db.session.add(namespaced_keyvalue)
+        db.session.commit()
+
+
+class NamespacedKeyValueView(restful.Resource):
+    parser = reqparse.RequestParser()
+
+    @auth.login_required
+    @requires_admin
+    @marshal_with(namespace_fields)
+    def get(self, namespace, key):
+        namespaced_keyvalue = NamespacedKeyValue.query.filter_by(namespace=namespace, key=key).first()
```

Comment:
Not exactly, we can also need all the keyvalue data for all the namespaces out there, just return an empty list if there is none, the same has been done for the blueprints and instances